example/poisson_gmg/poisson_package.hpp

```diff
@@ -0,0 +1,410 @@
+  using TE = parthenon::TopologicalElement;
+  auto pmb = md->GetBlockData(0)->GetBlockPointer();
+  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::entire, te);
+  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::entire, te);
+  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::entire, te);
+
+  int nblocks = md->NumBlocks();
+  std::vector<bool> include_block(nblocks, true);
+  if (only_interior) {
+    // The neighbors array will only be set for a block if its a leaf block
+    for (int b = 0; b < nblocks; ++b)
+      include_block[b] = md->GetBlockData(b)->GetBlockPointer()->neighbors.size() == 0;
+  }
+
+  if (only_md_level) {
+    for (int b = 0; b < nblocks; ++b)
+      include_block[b] =
+          include_block[b] &&
+          (md->grid.logical_level == md->GetBlockData(b)->GetBlockPointer()->loc.level());
+  }
+
+  auto desc = parthenon::MakePackDescriptor<a_t, b_t, out>(md.get());
+  auto pack = desc.GetPack(md.get(), include_block);
+  parthenon::par_for(
+      DEFAULT_LOOP_PATTERN, "SetPotentialToZero", DevExecSpace(), 0,
+      pack.GetNBlocks() - 1, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int b, const int k, const int j, const int i) {
+        pack(b, te, out(), k, j, i) =
+            wa * pack(b, te, a_t(), k, j, i) + wb * pack(b, te, b_t(), k, j, i);
+      });
+  return TaskStatus::complete;
+}
+
+template <class a_t, class b_t, class out, bool only_md_level = false>
+TaskStatus AddFieldsAndStore(std::shared_ptr<MeshData<Real>> &md, Real wa = 1.0,
+                             Real wb = 1.0) {
+  return AddFieldsAndStoreInteriorSelect<a_t, b_t, out, only_md_level>(md, wa, wb, false);
+}
+
+template <class var, bool only_md_level = false>
+TaskStatus SetToZero(std::shared_ptr<MeshData<Real>> &md) {
+  int nblocks = md->NumBlocks();
+  std::vector<bool> include_block(nblocks, true);
+  if (only_md_level) {
+    for (int b = 0; b < nblocks; ++b)
+      include_block[b] =
+          (md->grid.logical_level == md->GetBlockData(b)->GetBlockPointer()->loc.level());
+  }
+  auto desc = parthenon::MakePackDescriptor<var>(md.get());
+  auto pack = desc.GetPack(md.get(), include_block);
+  const size_t scratch_size_in_bytes = 0;
+  const int scratch_level = 1;
+  const int ng = parthenon::Globals::nghost;
+  parthenon::par_for_outer(
+      DEFAULT_OUTER_LOOP_PATTERN, "Print", DevExecSpace(), scratch_size_in_bytes,
+      scratch_level, 0, pack.GetNBlocks() - 1,
+      KOKKOS_LAMBDA(parthenon::team_mbr_t member, const int b) {
+        auto cb = GetIndexShape(pack(b, te, 0), ng);
+        const auto &coords = pack.GetCoordinates(b);
+        IndexRange ib = cb.GetBoundsI(IndexDomain::interior, te);
+        IndexRange jb = cb.GetBoundsJ(IndexDomain::interior, te);
+        IndexRange kb = cb.GetBoundsK(IndexDomain::interior, te);
+        parthenon::par_for_inner(
+            parthenon::inner_loop_pattern_simdfor_tag, member, kb.s, kb.e, jb.s, jb.e,
```

Comment:
Does this work on device?
I thought we only supported the simd one on the host and thus set the default one at compile time depending on the architecture.

Reply:
It at least compiles on device, but I haven't done extensive testing. IIRC, setting this to `DEFAULT_LOOP_PATTERN` won't compile anywhere.

Reply:
for a nested loop the defines are `DEFAULT_INNER_LOOP_PATTERN` and `DEFAULT_OUTER_LOOP_PATTERN` and overloads that set them automatically exists, i.e., `par_for_inner(member, ...` should work.

Reply:
ok, `DEFAULT_INNER_LOOP_PATTERN` seems to work. I thought I tried that before, but apparently not...

Reply:
hm, actually switching this to `DEFAULT_INNER_LOOP_PATTERN` fails to compile on device. I am not entirely sure why...

Reply:
I think this is a bug then. I'll open an issue.